release_notes/ocp-4-11-release-notes.adoc

```diff
@@ -1163,17 +1163,21 @@ The `haproxy.router.openshift.io/balance` variable, which sets the router load-b
 [id="ocp-4-11-legacy-service-account"]
 ==== LegacyServiceAccountTokenNoAutoGeneration is on by default
 
-To align with upstream link:https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#urgent-upgrade-notes-1[Kubernetes having moved] the `LegacyServiceAccountTokenNoAutoGeneration` feature gate to beta and enabling it by default, {product-title} now also follows this security feature and releases with the feature enabled. As a result, when creating new service accounts (SA), a service account token secret is no longer automatically generated. Previously, {product-title} automatically added a service account token to a secret for each new SA.
+In previous releases, two service account token secrets were generated when a service account was created:
 
-If you need a service account token secret, you must manually use the TokenRequest API to request bound service account tokens or create a service account token secret.
+* A service account token secret for authenticating to the internal {product-title} registry
+* A service account token secret for accessing the Kubernetes API
 
-After updating to {product-version}, existing service account token secrets are not deleted and continue to function as expected.
+Starting with {product-title} 4.11, this second service account token secret for accessing the Kubernetes API is no longer created. This is because the `LegacyServiceAccountTokenNoAutoGeneration` upstream Kubernetes feature gate was enabled in Kubernetes 1.24, which stops the automatic generation of secret-based service account tokens for accessing the Kubernetes API. After upgrading to {product-title} 4.11, any existing service account token secrets are not deleted and continue to function.
 
-Service Account token secrets still appear as auto-generated in {product-title} 4.11. However, instead of two secrets per Service Account, there will now be only one, which will be further reduced to zero in a future release. These tokens do not work. For now, dockercfg secrets are still being generated as secrets and no secrets will be deleted during upgrades.
+[NOTE]
+====
+Do not rely on these automatically generated secrets for your own use; they might be removed in a future {product-title} release.
+====
 
-* For information about using the TokenRequest API, see xref:../authentication/bound-service-account-tokens.html#bound-sa-tokens-configuring_bound-service-account-tokens[Using bound service account tokens]
+If your workload needs a service account token, request a bound service account token by using volume projection or by using the `TokenRequest` API directly. Tokens obtained from the `TokenRequest` API are more secure than service account token secrets, because they have a bounded lifetime and are not readable by other API clients. For more information, see xref:../authentication/bound-service-account-tokens.html#bound-sa-tokens-configuring_bound-service-account-tokens[Using bound service account tokens].
```

Comment:
> request a bound service account token by using volume projection

Users need not to do the request explicitly, because when a workload is created, it will automatedly create a bound service account token by using volume projection.

> by using the `TokenRequest` API directly

Instead of asking users to call the `TokenRequest` API directly, users could do it by command, the command calls  the `TokenRequest` API:
```shell
oc create token <serviceaccount name>
```

Reply:
@HanielZx what are you recommending we change this to then? Should we just more generically say something this?

```suggestion
If your workload needs a service account token, use bound service account tokens. Bound service account tokens are more secure than service account token secrets, because they have a bounded lifetime and are not readable by other API clients. For more information, see xref:../authentication/bound-service-account-tokens.html#bound-sa-tokens-configuring_bound-service-account-tokens[Using bound service account tokens].
```

Reply:
```suggestion
Each workload will be automatically injected with a projected volume to obtain a bound service account token. If your workload needs an additional service account token, add additional projected volume in your workload manifest. Bound service account tokens are more secure than service account token secrets, because they have a bounded lifetime and .... For more information, see xref:../authentication/bound-service-account-tokens.html#bound-sa-tokens-configuring_bound-service-account-tokens[Using bound service account tokens].
```
The part of "..." could be replaced by https://github.com/openshift/openshift-docs/pull/61541#discussion_r1250767436

Reply:
I ended up cutting out that whole section about why bound SA tokens are preferred over creating secrets, because that really isn't the important part of this release note entry (the main thing is to point out that we're not generating one of the secrets anymore). And we usually keep release notes shorter and link to elsewhere for more information. So the info on why they are preferred will exist (once my other PR is merged) in the "Using bound service account tokens" section that we link to.

So I think we are covered here - let me know if that sounds okay to you.